### PR TITLE
Qt/GameCubePane: Various fixes

### PR DESCRIPTION
**Pull request by spycrab**
Fixes the bug mentioned in [this forum post](https://forums.dolphin-emu.org/Thread-dolphin-qt-not-remembering-memory-card-location).
Should fix [issue #10848](https://bugs.dolphin-emu.org/issues/10848) as well.

### CODE REVIEW

File: Source/Core/DolphinQt2/Settings/GameCubePane.cpp [hunk 1]
```diff
@@ -253,6 +252,9 @@ void GameCubePane::LoadSettings()
     case ExpansionInterface::EXIDEVICE_MEMORYCARD:
       index = 2;
       break;
+    case ExpansionInterface::EXIDEVICE_ETH:
+      index = 2;
```
Comment (leoetlino) [inline Source/Core/DolphinQt2/Settings/GameCubePane.cpp]: Can we use named constants, or better, an enum? Looking at this made me realise how unclear the numbers are.

File: Source/Core/DolphinQt2/Settings/GameCubePane.cpp [hunk 2]
```diff
@@ -332,8 +337,6 @@ void GameCubePane::SaveSettings()
           dev,                    // The device enum to change to
           (index == 2) ? 2 : 0);  // SP1 is device 2, slots are device 0
     }
-
-    SConfig::GetInstance().m_EXIDevice[i] = dev;
```
Comment (leoetlino) [inline Source/Core/DolphinQt2/Settings/GameCubePane.cpp]: Could you please explain why this line was removed?

**Comment by spycrab** [comment]
@dolphin-emu-bot rebuild

File: Source/Core/DolphinQt2/Settings/GameCubePane.cpp [hunk 3]
```diff
@@ -302,25 +314,28 @@ void GameCubePane::SaveSettings()
...
       dev = ExpansionInterface::EXIDEVICE_NONE;
       break;
-    case 1:
+    case EXP_DUMMY:
       dev = ExpansionInterface::EXIDEVICE_DUMMY;
       break;
-    case 2:
-      dev = ExpansionInterface::EXIDEVICE_MEMORYCARD;
+    case EXP_MEMORYCARD:
+      if (i == 3)
```
Comment (BhaaLseN) [inline Source/Core/DolphinQt2/Settings/GameCubePane.cpp]: Could you please leave a comment why `i == 3` is special?

Comment (spycrab) [reply Source/Core/DolphinQt2/Settings/GameCubePane.cpp]: I think it's clearer now, also the magic number was wrong anyway :/ (i < 3 loops should never reach 3 :smile:)

**Comment by spycrab** [comment]
@dolphin-emu-bot rebuild

**Pull request merged**